File: ui/src/models/document.ts
export enum DocumentStatus {
  PROCESSING = "PROCESSING",
  COMPLETE = "COMPLETE",
  FAILED = "FAILED",
}

export interface Document {
  id: string;
  filename: string;
  status: DocumentStatus;
}

export interface PaginatedDocuments {
  documents: Document[];
  total: number;
  offset: number;
  take: number;
}

File: api/app/api/models/document.py
from typing import List, Optional
from uuid import UUID

from . import BaseSchema
from .document_status import DocumentStatus


class DocumentResponse(BaseSchema):
    id: UUID
    filename: str
    status: DocumentStatus


class PaginatedDocumentResponse(BaseSchema):
    documents: List[DocumentResponse]
    total: int
    offset: int
    take: int
